clustered/integration-test/src/test/java/org/ehcache/clustered/reconnect/BasicCacheReconnectTest.java

```diff
@@ -0,0 +1,128 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import static org.ehcache.clustered.util.TCPProxyUtil.setDelay;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.terracotta.testing.rules.BasicExternalClusterBuilder.newCluster;
+
+public class BasicCacheReconnectTest {
+  private static final String RESOURCE_CONFIG =
+          "<config xmlns:ohr='http://www.terracotta.org/config/offheap-resource'>"
+                  + "<ohr:offheap-resources>"
+                  + "<ohr:resource name=\"primary-server-resource\" unit=\"MB\">64</ohr:resource>"
+                  + "</ohr:offheap-resources>"
+                  + "</config>\n"
+                  + "<service xmlns:lease='http://www.terracotta.org/service/lease'>"
+                  + "<lease:connection-leasing>"
+                  + "<lease:lease-length unit='seconds'>5</lease:lease-length>"
+                  + "</lease:connection-leasing>"
+                  + "</service>";
+
+  @ClassRule
+  public static Cluster CLUSTER =
+          newCluster().in(new File("build/cluster")).withServiceFragment(RESOURCE_CONFIG).build();
+
+  @BeforeClass
+  public static void waitForActive() throws Exception {
+    CLUSTER.getClusterControl().waitForActive();
+  }
+
+  private final List<TCPProxy> proxies = new ArrayList<>();
+
+  @Test
+  public void cacheOpsDuringReconnection() throws Exception {
+
+    URI connectionURI = TCPProxyUtil.getProxyURI(CLUSTER.getConnectionURI(), proxies);
+
+    CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder
+            = CacheManagerBuilder.newCacheManagerBuilder()
+            .with(ClusteringServiceConfigurationBuilder.cluster(connectionURI.resolve("/crud-cm"))
+                    .autoCreate()
+                    .defaultServerResource("primary-server-resource"));
+    PersistentCacheManager cacheManager = clusteredCacheManagerBuilder.build(false);
+    cacheManager.init();
+
+    CacheConfiguration<Long, String> config = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class,
+            ResourcePoolsBuilder.newResourcePoolsBuilder()
+                    .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)))
+            .withResilienceStrategy(new ThrowingResiliencyStrategy<>())
+            .build();
+
+    Cache<Long, String> cache = cacheManager.createCache("clustered-cache", config);
+
+    CompletableFuture<Void> future = CompletableFuture.runAsync(() ->
+            ThreadLocalRandom.current()
+            .longs()
+            .forEach(value ->
+                    cache.put(value, Long.toString(value))));
+
+    setDelay(6000, proxies);
```

Comment:
More delay I would like to remove

Reply:
Can't do that, since any time less than this does not leave enough time for reconnect to happen in test environment. I think we will have to live with this, These tests are here just for basic integration coverage

Reply:
Fair enough. That's something we might want to address in TCServer